lib/request_handler/schema_parser.rb

```diff
@@ -17,22 +17,22 @@ def initialize(schema:, schema_options: {})
 
     def validate_schema(data, with: schema)
       raise MissingArgumentError, data: 'is missing' if data.nil?
-      data = deep_symbolize_keys(data) if with.rules.keys.first.is_a?(Symbol)
+      if RequestHandler.backend == Validation::DryBackend
```

Comment:
how about hiding this inside the backend

Reply:
no this is somehow needed only in this special case in the schema parser... I don't quite understand why that is done but according to @sarahdiendorfer its bescause of something in samples?

Reply:
hmm I assume a outdated version of something causes this.

Can you try to find why this is needed. Add a test around it OR remove this special handling?

Reply:
ok there is exactly one spec for which this is needed, which is https://github.com/runtastic/request_handler/blob/master/spec/request_handler/schema_parser_spec.rb#L112-L134

It seems that this is a hack to fix some unexpected dry-validation behaviour. The thing is that in this test fails with:

```
 expected: {:nested=>{:attr1=>"a1", :attr2=>"a2"}, :simple=>5}
           got: {:nested=>{"attr1"=>"a1", "attr2"=>"a2"}, :simple=>5}
```
As you can see the subhash is expected to have symbol_keys but dry does for some weird reason only convert the top level keys, even though the specified schema is a params schema...

So I see two options:

1. We always deep symbolize the key of every input to have a consistent behaviour. That might break usage where we specified string keys via our schemas though (but i doubt that we do that anywhere?)
2. We leave this hack as is to support both symbol and string key output for the dryEngine

FYI the DefinitionEngine already symbolizes all input, otherwise we would need to either use string key output with definition OR add a feature to definition that does transform the keys if configured (which would be a good feature anyways)

Reply:
ah I remember now. the issue is that you can define nested dry validation schemas were the inner keys are strings. it will somehow automatically deal with the top level keys but not the nested keys.

I still feel like this workaround should be moved inside the engine and not only done here - possible that this is actually a bug now in other cases where this is not done

Reply:
ok. then i leave it in. Its already moved to the engine now